include/Engine/Utility/Config/ConfigSection.h

```diff
@@ -40,6 +41,43 @@ namespace gp1 {
 			// Get T value of a config.
 			template <typename T>
 			T GetConfigTyped(const std::string& key, T def);
+			// Sets a config value with an enum value name.
+			template <typename T>
+			void SetConfigEnum(const std::string& key, T value, const std::vector<std::pair<T, std::string>>& enumNames) {
```

Comment:
Maybe we can make this a bit shorter by adding an alias? Something like
```cpp
template <typename T> using EnumVector = std::vector<std::pair<T, std::string>>;
```
that we later can use for `InputNames` etc!

Reply:
Yeah that sounds good. 😄